### ISSUE TITLE
Encrypted files can't be opened with external zip software

### ISSUE THREAD

**Issue by s-weigand** [danifus/pyzipper#2]
When I create a zip file with example code from the frontpage (just different password), reading it works fine python. 

```
In [1]: import pyzipper
      :
      : secret_password = b'test123'
      :
      : with pyzipper.AESZipFile('new_test.zip',
      :                          'w',
      :                          compression=pyzipper.ZIP_LZMA,
      :                          encryption=pyzipper.WZ_AES) as zf:
      :     zf.pwd = secret_password
      :     zf.writestr('test.txt', "What ever you do, don't tell anyone!")
      :
      : with pyzipper.AESZipFile('new_test.zip') as zf:
      :     zf.pwd = secret_password
      :     my_secrets = zf.read('test.txt')
      :     print(my_secrets)
      :
b"What ever you do, don't tell anyone!"
```

But when I try to unpack it with zip software (WinRaR or 7zip) it says "Wrong password".

**OS**
Windows 10

**Software**
- WinRaR 5.61 (x64)
- 7zip 18.06 (x64)

**pyzipper version**
- pyzipper from pip + `pycrypto`
- #1 pyzipper +  `pycryptodomex`

PS.: In the projects i want to use `pyzipper` i want to bundel and encrypt config files, which are than only read by python again, so it kinda is an extra security feature. But still wanted to mention this issue, maybe keep it with a `python_only` flag/kwarg 😄

**Comment by danifus** [comment]
I've found the cause of the issue (some over eager refactoring) and fixed it locally. I'm just adding some tests and will push it later today

**Comment by danifus** [comment]
haha My goal was compatibility with other zip programs. It's a pretty small fix so you could reimplement it if you wanted ;)

Fixed in d504e57c3e1a2c8dff8feb727133c025409b8f43
